java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1806,6 +1806,25 @@ public ColumnVector stringLocate(Scalar substring, int start, int end) {
     }
   }
 
+  /**
+   * Returns a list of columns by splitting each string using the specified delimiter.
+   * The number of rows in the output columns will be the same as the input column.
+   * Null entries are added for a row where split results have been exhausted.
+   * Null string entries return corresponding null output columns.
+   * @param delimiter UTF-8 encoded string indentifying the split points in each string.
+   *                  Default of empty string indicates split on whitespace.
+   * @return New table of strings columns.
+   */
+  public Table stringSplit(Scalar delimiter) {
```

Comment:
What happens if the delimiter is null?

Reply:
Added the assert for the case when delimiter is null.

Reply:
Does it make sense to provide an API that takes in a Java String instead of a Scalar? As a user I find it hard to having to first create a scalar only to close it.

Reply:
cudf expects delimiter to be string_scalar. So keeping the same here to be in sync. Please let me know if it's okay.